cfme/common/provider.py

```diff
@@ -766,3 +787,40 @@ def import_all_modules_of(loc):
     path = project_path.join('{}'.format(loc.replace('.', '/'))).strpath
     for _, name, _ in pkgutil.iter_modules([path]):
         importlib.import_module('{}.{}'.format(loc, name))
+
+
+
+@navigator.register(ProviderList, 'All')
+class ProviderListing(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn(*self.obj.NAVTREE_LOCATION)(None)
+
+    def resetter(self):
+        # Reset view and selection
+        tb.select("Grid View")
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
+
+
+@navigator.register(BaseProvider, 'All')
+class All(CFMENavigateStep):
+    @property
+    def prereqisite(self):
+        if isinstance(self.ob, type):
+            return self.prereqisite_type
+        else:
+            return self.prerequisite_instance
+
+    prerequisite_instance = NavigateToAttribute('collection', 'All')
```

Comment:
This basically gives us the backwards compat, hmmm it's a little more work on the navmazing side of things, a little more confusing too, I was actually wondering if we don't jsut say to people, you can't call All on a Provider, it has to be on the Collection object.....hmmm.....thinking....